src/traces/scatter/hover.js

```diff
@@ -107,16 +107,16 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
                 // always will show up regardless of point size, but
                 // prioritize smaller points
                 var rad = Math.max(3, di.mrc || 0);
-                return Math.max(Math.abs(xa.c2p(di.x) - xa.c2p(xval)) - rad, 1 - 3 / rad);
+                return Math.max(Math.abs(xa.c2p(di.x) - xpx) - rad, 1 - 3 / rad);
             },
             dy = function(di) {
                 var rad = Math.max(3, di.mrc || 0);
-                return Math.max(Math.abs(ya.c2p(di.y) - ya.c2p(yval)) - rad, 1 - 3 / rad);
+                return Math.max(Math.abs(ya.c2p(di.y) - ypx) - rad, 1 - 3 / rad);
             },
             dxy = function(di) {
                 var rad = Math.max(3, di.mrc || 0),
-                    dx = Math.abs(xa.c2p(di.x) - xa.c2p(xval)),
-                    dy = Math.abs(ya.c2p(di.y) - ya.c2p(yval));
+                    dx = xa.c2p(di.x) - xpx,
+                    dy = ya.c2p(di.y) - ypx;
```

Comment:
`dx`, `dy`, and `dxy` are the hot path for scatter hover, this should make them a little faster.